burnman/nonlinear_fitting.py

```diff
@@ -0,0 +1,314 @@
+import numpy as np
+from scipy.stats import t
+import itertools
+    
+def nonlinear_least_squares_fit(model,
+                                mle_tolerance,
+                                lm_damping = 0.,
+                                param_tolerance = 1.e-7,
+                                max_lm_iterations = 100,
+                                verbose = False):
+
+    """
+    Function to compute the "best-fit" parameters for a model
```

Comment:
you are replacing ``scipy.optimize.curve_fit`` by this. Can you explain what you are doing differently now? Is this still a Levenberg-Marquardt?